vendor/share-modal.css

```diff
@@ -112,4 +96,49 @@
 .share-modal input[type=text] {
   width: calc(100% - 120px);
   margin-left: 0;
-}
\ No newline at end of file
+}
+
+.share-modal .permission-list {
+  margin: 0;
+  margin-left: -10px;
+  padding: 0;
+}
+
+.share-modal .permission-list .permission-item {
+  color: #333;
+  display: inline-block;
+  margin: 5px 10px;
+  
+  border: 1px solid #DDD;
+  -webkit-border-radius: 2px;
+     -moz-border-radius: 2px;
+      -ms-border-radius: 2px;
+          border-radius: 2px;
+  
+  padding: 10px;
+  box-sizing:border-box;
+  text-align: center;    
+  cursor: default;
+}
+
+.share-modal .permission-list .permission-item span {
+  vertical-align: middle;
+}
+
+.share-modal .permission-list .permission-item a {
+  vertical-align: middle;
+  font-weight: bold;
+  cursor: pointer;
+  color: gray;
+  margin-left: 5px;
+}
+
+.share-modal .permission-list .permission-item a:hover {
+  color: black;
+}
+
+.share-modal .permission-list .permission-item a:before {
+  display: inline-block;
+  content: 'x';
```

Comment:
I'm using a literal 'x' character here, since that helps as avoid including stuff like icon-fonts in the project. I figured with the correct coloring, it can look good enough for a basic style. If the add-on user would prefer a different look, they can always override it.